lib/web/fetch/webidl.js

```diff
@@ -354,12 +354,9 @@ webidl.interfaceConverter = function (i) {
 
 webidl.dictionaryConverter = function (converters) {
   return (dictionary, prefix, argument) => {
-    const type = webidl.util.Type(dictionary)
     const dict = {}
 
-    if (type === 'Null' || type === 'Undefined') {
-      return dict
-    } else if (type !== 'Object') {
+    if (dictionary != null && typeof dictionary !== 'object') {
```

Comment:
```suggestion
    if (dictionary != null && webidl.util.Type(dictionary) !== 'Object') {
```

The 'function' is also an 'Object' but it errors.